salt/grains/core.py

```diff
@@ -2200,52 +2190,12 @@ def fqdns():
     '''
     Return all known FQDNs for the system by enumerating all interfaces and
     then trying to reverse resolve them (excluding 'lo' interface).
+    To disable the fqdns grain, set enable_fqdns_grains: False in the minion configuration file.
     '''
-    # Provides:
-    # fqdns
-
-    grains = {}
-    fqdns = set()
-
-    def _lookup_fqdn(ip):
-        try:
-            name, aliaslist, addresslist = socket.gethostbyaddr(ip)
-            return [socket.getfqdn(name)] + [als for als in aliaslist if salt.utils.network.is_fqdn(als)]
-        except socket.herror as err:
-            if err.errno in (0, HOST_NOT_FOUND, NO_DATA):
-                # No FQDN for this IP address, so we don't need to know this all the time.
-                log.debug("Unable to resolve address %s: %s", ip, err)
-            else:
-                log.error(err_message, err)
-        except (socket.error, socket.gaierror, socket.timeout) as err:
-            log.error(err_message, err)
-
-    start = time.time()
-
-    addresses = salt.utils.network.ip_addrs(include_loopback=False, interface_data=_get_interfaces())
-    addresses.extend(salt.utils.network.ip_addrs6(include_loopback=False, interface_data=_get_interfaces()))
-    err_message = 'Exception during resolving address: %s'
-
-    # Create a ThreadPool to process the underlying calls to 'socket.gethostbyaddr' in parallel.
-    # This avoid blocking the execution when the "fqdn" is not defined for certains IP addresses, which was causing
-    # that "socket.timeout" was reached multiple times secuencially, blocking execution for several seconds.
-
-    try:
-       pool = ThreadPool(8)
-       results = pool.map(_lookup_fqdn, addresses)
-       pool.close()
-       pool.join()
-    except Exception as exc:
-       log.error("Exception while creating a ThreadPool for resolving FQDNs: %s", exc)
-
-    for item in results:
-        if item:
-            fqdns.update(item)
-
-    elapsed = time.time() - start
-    log.debug('Elapsed time getting FQDNs: {} seconds'.format(elapsed))
-
-    return {"fqdns": sorted(list(fqdns))}
+    opt = {"fqdns": []}
+    if __opts__.get('enable_fqdns_grains', True) == True:
```

Comment:
Do we really want to have the "fqdns" calculation in the grains by default on Salt, regardless of SUMA?

I would instead set the default behavior to `False` (not including fqdns in grains) and, in case this grain is still needed by SUMA, then explicitly add `enable_fqdns_grains` only on the SUSE Manager context.

@moio, @dincamihai - what do you think?

Reply:
Isn't the current behavior to include the fqdns grains?

Reply:
Yes, that is the current behavior. But we want to get rid of that because the unpredictable problems when the network configuration is wrong and core grains takes too long into calculate.

Reply:
From here: https://github.com/SUSE/spacewalk/issues/8428#issuecomment-517565935
The idea was that we don't change the default behavior with this PR but we allow SUSE Manager to disable the fqdns grains with a config param.

Reply:
Idea was to be nice at upstream users